server/blob/blob-union/src/test/java/org/apache/james/blob/union/UnionBlobStoreTest.java

```diff
@@ -224,6 +247,61 @@ void readBytesShouldReturnFallbackToLegacyWhenCurrentCompletedExceptionally() th
         }
     }
 
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
```

Comment:
normally `@MethodSource` - parameter factory, only able to access static method. I need to declare `CurrentAndLegacyCouldNotComplete` test instance as `PER_CLASS`. This should help `@MethodSource` access to non static method factory. see [junit 5 documentaion](https://junit.org/junit5/docs/5.2.0/user-guide/#writing-tests-parameterized-tests-sources-MethodSource)